src/components/TasksList/TasksList.tsx

```diff
@@ -27,13 +93,53 @@ const TasksList: React.FC = () => {
           <Tab label='My tasks' />
         </Tabs>
       </AppBar>
-      <TabPanel value={value} index={0}>
-        Newbie Tasks Component
+
+      {updateTaskLoading && <LinearProgress />}
+
+      <TabPanel p={2} value={tabIndex} index={0}>
+        {loading && <PlaceHolderTaskList />}
+        {!loading && data && data.newbie.newbieTasks.length === 0 && (
+          <EmptyTaskList />
+        )}
+        {!loading && data && data.newbie.newbieTasks.length > 0 && (
+          <TaskTabsContent
+            onChange={onTaskChange}
+            tasks={data.newbie.newbieTasks as Task[]}
+          />
+        )}
       </TabPanel>
-      <TabPanel value={value} index={1}>
-        My Tasks Component
+      <TabPanel p={2} value={tabIndex} index={1}>
```

Comment:
In addition to my previous post, I think that original `<TabPanel>` could be extended to hide all those conditionals.
Currently it's a very simple wrapper, but it may consume some props and take care of all that complex logic.

Reply:
^